src/consumer/kafka.py

```
@@ -0,0 +1,237 @@
+        
+        """
+        
+        # Set up signal handlers for graceful shutdown
+        
+        loop = asyncio.get_event_loop()
+        for sig in (signal.SIGTERM, signal.SIGINT):
+            try:
+                loop.add_signal_handler(sig, lambda s=sig: asyncio.create_task(self._handle_shutdown(s)))
+            except NotImplementedError:
+                # Fallback for Windows / non-main thread
+                signal.signal(sig, lambda *_: asyncio.create_task(self._handle_shutdown(sig)))
```

Ruff (0.8.2)

Function definition does not bind loop variable `sig`

(B023)